zookeeper-server/src/main/java/org/apache/zookeeper/server/NettyServerCnxnFactory.java

```diff
@@ -316,16 +251,17 @@ public void operationComplete(ChannelFuture future)
                     if (KeeperException.Code.OK !=
                             authProvider.handleAuthentication(cnxn, null)) {
                         LOG.error("Authentication failed for session 0x{}",
-                                Long.toHexString(cnxn.sessionId));
+                                Long.toHexString(cnxn.getSessionId()));
                         cnxn.close();
                         return;
                     }
 
-                    allChannels.add(future.getChannel());
+                    final Channel futureChannel = future.getNow();
```

Comment:
I think `get()` would be enough, but the check is harmful anyway.

Reply:
I think they are equivalent here since we know the future is completed (we are inside a `if (future.isSuccess())` block). I can change it to `get()` if you prefer.